types/index.d.ts

```diff
@@ -270,7 +270,46 @@ declare class PASETOStructured {
   payload: AnyObject;
 }
 
-declare class BaseToken {
+interface BaseModel {
+  jti: string;
+  kind: string;
+  iat?: number;
+  exp?: number;
+}
+
+declare class BaseModel {
+  get adapter(): Adapter;
+
+  save(ttl?: number): Promise<string>;
+  destroy(): Promise<void>;
+  emit(eventName: string): void;
+
+  static get adapter(): Adapter;
+
+  static get IN_PAYLOAD(): string[];
+
+  static find<T>(
+    this: { new (...args: any[]): T },
+    id: string,
+    options?: object
+  ): Promise<T | undefined>;
+
+  /* ----- maxin from has_format ----- */
+  // see https://github.com/panva/node-oidc-provider/blob/master/lib/models/mixins/has_format.js
+
+  generateTokenId(): string;
+  getValueAndPayload(): [string, object];
+  getTokenId(token: string): string;
+  verify<Payload>(
+    token: string,
+    payload: Payload,
+    args: { ignoreExpiration: boolean; foundByReference: boolean }
+  ): Payload;
+
+  // ---------------------------------------- //
```

Comment:
This whole has_format block is NOT a public API anymore since v6.x, please REMOVE.

Reply:
reverted ;)

Reply:
Please remove `generateTokenId`, `getValueAndPayload`, `getTokenId`, and `verify`.

Reply:
done